confluent_kafka/src/Producer.c

```diff
@@ -251,7 +293,8 @@ static PyObject *Producer_produce (Handle *self, PyObject *args,
 	int value_len = 0, key_len = 0;
 	int partition = RD_KAFKA_PARTITION_UA;
 	PyObject *dr_cb = NULL, *dr_cb2 = NULL, *partitioner_cb = NULL;
-	rd_kafka_topic_t *rkt;
+        long long timestamp = 0;
```

Comment:
re: indent: I see, on GH I guess it becomes obvious due to the extra `+` in the diff that the tabs on the previous line don't account for. I hate losing git history, but there's really no good time to do a mass reformatting, so we should probably just bite the bullet and do it once we hit a quiet period with PRs.

re: timestamps, I think we end up diverging just mildly from Java behavior. For some reason I had it in my head that if you continued to pass null to the Java producer it would fallback to LogAppendTime, but apparently newer clients will default to the current time on the client if the timestamp is null. Since we can't do that with a `long long`, the `0` sentinel value works fine -- I suppose not many folks will be trying to use the epoch as a real timestamp...